clients/src/test/java/io/strimzi/integration/KafkaClientWithRegistryIT.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.integration;
+
+import io.apicurio.registry.rest.v2.beans.IfExists;
+import io.apicurio.registry.serde.SerdeConfig;
+import io.apicurio.registry.serde.avro.AvroKafkaDeserializer;
+import io.apicurio.registry.serde.avro.AvroKafkaSerializer;
+import io.apicurio.registry.serde.config.IdOption;
+import io.skodjob.datagenerator.enums.ETemplateType;
+import io.strimzi.configuration.ConfigurationConstants;
+import io.strimzi.kafka.KafkaConsumerClient;
+import io.strimzi.kafka.KafkaProducerClient;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.config.TopicConfig;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Stream;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class KafkaClientWithRegistryIT extends AbstractIT {
+    private static GenericContainer registry;
+
+    @BeforeAll
+    void startRegistry() {
+        registry = new GenericContainer<>(DockerImageName.parse("quay.io/apicurio/apicurio-registry-mem:2.6.5.Final"))
```

Comment:
Should this be in some constant?

Reply:
Ye, I will also try to bump version to 3.x which seems to be latest available.